includes/printf.h

```diff
@@ -0,0 +1,14 @@
+#ifndef PRINTF_H
+#define PRINTF_H
+
+#include <stddef.h>
+
+typedef struct  writer_s
```

Comment:
nitpick. consider moving writer interface to it's own header. There could be other users of this interface beside printf